### DISCUSSION TITLE
SQL Server; DB-Lib error message 20009, severity 9

### DISCUSSION THREAD

**Discussion by gauiPPP** [getredash/redash#6398] [Q&A]
I am getting this error message after 60 seconds when trying to query MS SQL server:
"Error running query: DB-Lib error message 20009, severity 9: Unable to connect: Adaptive Server is unavailable or does not exist ([server IP address]:1433) Net-Lib error during Connection timed out (110)"
Also "test connection" fails with timeout under data source.

This started on 16/05/23 01:33 or that is the timestamp for last executed script. 
No changes were made, as far as I can tell. Scheduled updates for the ubuntu environment were on the next day.

I have installed redash with the provided installation script onto Ubuntu 18.04 LTS, running on a virtual machine. Redash worked without problems for several months. Of course I had not VM backups / snapshots running in proxmox (now they are) so I cannot revert back to a time before problems starting.
Pinging the SQL server and port from the ubuntu console works.

Is there something to be done or should I try simply a new installation from scratch? Can I somehow import the queries etc from the current setup into the new?

**Comment by gauiPPP** [comment]
Redash was running with version 8.0. I now upgraded with these instructions https://github.com/getredash/redash/releases/tag/v10.0.0 to 10.0 and then to 10.1. These upgrades did not help with the issue. During the update process I had a view on the status of workers, there I saw the data source looping and getting an error.

**Comment by guidopetri** [comment]
- are you able to access the machine running Redash?
- are you able to access the SQL server directly?
- is the Redash machine able to access the SQL server (outside of Redash)?
- did the SQL server have any updates around when the problem started happening?

To me this sounds like a networking problem, especially given its suddenness. Maybe a new firewall rule got implemented?

**Reply by gauiPPP** [reply]
I am able to access ubuntu machine and also the redash web UI.
SQL server can be accessed with the same credentials as are being used by redash. On ubuntu the command telnet [SQL server] 1433 opens a connection.
The SQL server had updates a week before but not on that date. Later we have upgraded the SQL server Windows Server and the MS SQL server to newer versions. Everything else works with the newer OS and SQL server but not redash.
